src/build.rs

```diff
@@ -519,9 +535,66 @@ pub fn compile_options(
         FilterRule::none(),
     );
 
+    compile_opts.build_config.unit_graph = false;
+
     Ok(compile_opts)
 }
 
+#[derive(Default)]
+struct Exec {
+    ran: AtomicBool,
+    link_line: Mutex<String>,
+}
+
+use cargo::core::*;
+use cargo::util::ProcessBuilder;
+use cargo::CargoResult;
+
+impl Executor for Exec {
+    fn exec(
+        &self,
+        cmd: &ProcessBuilder,
+        _id: PackageId,
+        _target: &Target,
+        _mode: CompileMode,
+        on_stdout_line: &mut dyn FnMut(&str) -> CargoResult<()>,
+        on_stderr_line: &mut dyn FnMut(&str) -> CargoResult<()>,
+    ) -> CargoResult<()> {
+        self.ran.store(true, Ordering::Relaxed);
+        cmd.exec_with_streaming(
+            on_stdout_line,
+            &mut |s| {
+                #[derive(serde::Deserialize, Debug)]
+                struct Message {
+                    message: String,
+                    level: String,
+                }
+
+                if let Ok(msg) = serde_json::from_str::<Message>(s) {
+                    // suppress the native-static-libs messages
+                    if msg.level == "note" {
+                        if msg.message.starts_with("Link against the following native artifacts when linking against this static library") {
+                            Ok(())
+                        } else if msg.message.starts_with("native-static-libs:") {
+                            let (_, link_line) = msg.message.split_at(20);
```

Comment:
Annoying that github does not allow to add suggestions when reviewing individual commits...

```rust
} else if let Some(link_line) = msg.message.strip_prefix("native-static-libs:") {
```

Reply:
thank you :)